plugin.go

```diff
@@ -142,140 +131,245 @@ func meta(environment, tags string, options metaOptions, k keyring.Keyring) erro
-		packageErr = packageCmd.Run()
-		if packageErr != nil {
-			return
+		// Get repo name
+		repoName, err := getRepoName()
+		if err != nil {
+			log.Fatal("Failed to get repo name: %v", err)
 		}
 
-		publishCmdArgs := []string{"publish"}
-		publishCmdArgs = append(publishCmdArgs, commonArgs...)
-		publishCmd := keyringCmd(plasmaBinary, publishCmdArgs...)
-		publishCmd.Stdout = &publishStdOut
-		publishCmd.Stderr = &publishStdErr
-		//publishCmd.Stdin = os.Stdin // Any interaction will prevent waitgroup to finish and thus stuck before print of stdout
-		//cli.Println(sanitizeString(publishCmd.String(), keyringPassphrase)) // TODO: Debug why it appears during deploy command stdout
-		publishErr = publishCmd.Run()
-		if publishErr != nil {
-			return
+		// Get project ID
+		projectID, err := getProjectID(gitlabDomain, username, password, accessToken, repoName)
+		if err != nil {
+			log.Fatal("Failed to get project ID: %v", err)
 		}
-	}(wg)
-
-	wg.Add(1)
-	go func(wg *sync.WaitGroup) {
-		defer wg.Done()
-		deployCmdArgs := []string{"platform:deploy"}
-		deployCmdArgs = append(deployCmdArgs, environment)
-		deployCmdArgs = append(deployCmdArgs, tags)
-		if verbosity != "" {
-			deployCmdArgs = append(deployCmdArgs, "--debug")
+
+		// Trigger pipeline
+		pipelineID, err := triggerPipeline(gitlabDomain, username, password, accessToken, projectID, branchName, environment, tags, comparisonRef)
+		if err != nil {
+			log.Fatal("Failed to trigger pipeline: %v", err)
 		}
 
-		deployCmd := keyringCmd(plasmaBinary, deployCmdArgs...)
-		deployCmd.Stdout = os.Stdout
-		deployCmd.Stderr = os.Stderr
-		deployCmd.Stdin = os.Stdin
-		cli.Println(sanitizeString(deployCmd.String(), options.keyringPassphrase))
-		deployErr := deployCmd.Run()
-		if deployErr != nil {
-			handleCmdErr(deployErr)
+		// Get all jobs in the pipeline
+		jobs, err := getJobsInPipeline(gitlabDomain, username, password, accessToken, projectID, pipelineID)
+		if err != nil {
+			log.Fatal("Failed to retrieve jobs in pipeline: %v", err)
 		}
-	}(wg)
-	wg.Wait()
-
-	fmt.Println()
-	fmt.Println(packageStdOut.String())
-	fmt.Println(packageStdErr.String())
-	if packageErr != nil {
-		handleCmdErr(packageErr)
-	}
 
-	fmt.Println(publishStdOut.String())
-	fmt.Println(publishStdErr.String())
-	if publishErr != nil {
-		handleCmdErr(publishErr)
-	}
+		// Find the target job ID
+		var targetJobID int
+		for _, job := range jobs {
+			if job.Name == targetJobName {
+				targetJobID = job.ID
+				break
+			}
+		}
+		if targetJobID == 0 {
+			log.Fatal("No %s job found in pipeline", targetJobName)
+		}
+
+		// Trigger the manual job
+		err = triggerManualJob(gitlabDomain, username, password, accessToken, projectID, targetJobID, pipelineID)
+		if err != nil {
+			log.Fatal("Failed to trigger manual job: %v", err)
+		}
+
+	} else {
+		cli.Println("Starting local build...")
```

Comment:
splitting these 2 chunks of code onto 2 function, like CIbuild, localBuild would benefit readability